### ISSUE TITLE
Newline in generated reference URL

### ISSUE THREAD

**Issue by magnusja** [juhaku/utoipa#1019]
Hey there,

thanks for this library first of all. 

For some weird reason I have a newline in a reference URL.
```
#[utoipa::path(
get,
path = "/calculations/assembly-costs",
params(CalculationAssemblyCostParams),
responses(
(status = 200, description = "Get calculated cost of an assembly.",
    body = types::calculation::calculation_assembly_cost::v1::CalculationAssemblyCostResponse)
),
)]
```

yields
```
"/calculations/assembly-costs": {
      "get": {
        "tags": [
          "routes::calculation_assembly_cost"
        ],
        "operationId": "get_single_assembly_cost_v1",
        "parameters": [
          {
            "name": "assembly_id",
            "in": "query",
            "required": true,
            "schema": {
              "type": "string",
              "format": "uuid"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Get calculated cost of an assembly.",
            "content": {
              "application/json": {
                "schema": {
                  /// BAD
                  "$ref": "#/components/schemas/types.calculation.calculation_assembly_cost.v1 ::\nCalculationAssemblyCostResponse"
                }
              }
            }
          }
        }
      }
    },
```

Unfotunately this throws off the swagger editor. But I really dont understand why the newline is added there in the first place?

**Comment by juhaku** [comment]
Hmm, super interesting, There shouldn't be new line though. This is something that needs to be investigated. What version of `utoipa` you are using?

However the way the body reference names are going to be resolved in future will change in the new 5.0.0 release once it is out. The path will not end up defining the name anymore. But it will be solely controlled with `as` attribute if any prefixing is needed & wanted.

**Comment by juhaku** [comment]
This seems to be issue in the latest `rc` as well. I try to fix this. Also it seems that it generates the name for the type incorrectly by the `as` attribute.
